CONTRIBUTING.md

```diff
@@ -0,0 +1,59 @@
+# Contributing
+
+Thanks for considering contributing!  We want AllenNLP to be *the way* to do cutting-edge NLP research, but we cannot
+get there without community support.
+
+## How Can I Contribute?
+
+### Did you find a bug?
+
+First, do [a quick search](https://github.com/allenai/allennlp/issues) to see whether your issue has already been reported.
+If your issue has already been reported, please comment on the existing issue.
+
+Otherwise, open [a new GitHub issue](https://github.com/allenai/allennlp/issues).  Be sure to include a clear title
+and description.  The description should include as much relevant information as possible.  The description should
+explain how to reproduce the erroneous behavior as well as the behavior you expect to see.  Ideally you would include a
+code sample or an executable test case demonstrating the expected behavior.
+
+### Did you write a fix for a bug?
+
+Open [a new GitHub pull request](https://github.com/allenai/allennlp/pulls) with the fix.  Make sure you have a clear
+description of the problem and the solution, and include a link to relevant issues.
+
+Once your pull request is created, our continuous build system will check your pull request.  Continuous
+build will test that:
+
+* The code compiles
+* All tests pass
+* `pylint` accepts the code style (our guidelines are based on PEP8)
+* `mypy` typechecks the Python code
```

Comment:
@matt-gardner I'd like to add something here about our code coverage checks.  Where can I find the rules those follow presently?